fiftyone/core/labels.py

```diff
@@ -2103,3 +2188,547 @@ def _from_geo_json(d):
+                logger.warning(
+                    "Skipping SMPLHumanPose %d with no vertices", idx
+                )
+                continue
+
+            person_id = (
+                person.person_id if person.person_id is not None else idx
+            )
+            vertices = np.asarray(person.vertices)
+            all_vertices.append(vertices)
+
+            mesh_filename = f"human_mesh_{uid}_person_{person_id}.obj"
+            mesh_path = os.path.join(base_dir, mesh_filename)
+
+            mesh = trimesh.Trimesh(
+                vertices=vertices, faces=smpl_faces_arr, process=False
+            )
+            mesh.export(mesh_path)
+
+            mesh_objects.append(
+                ObjMesh(
+                    name=f"Person {person_id}",
+                    obj_path=mesh_path,
+                )
+            )
+
+        if not mesh_objects:
+            raise ValueError("Cannot export scene: no meshes were generated")
+
+        all_vertices = np.vstack(all_vertices)
+        center = all_vertices.mean(axis=0)
+        bbox_size = all_vertices.max(axis=0) - all_vertices.min(axis=0)
+        camera_distance = bbox_size.max() * 1.5
+        camera_position = center + np.array([0, 0, camera_distance])
+
+        aspect = (
+            frame_size[1] / frame_size[0]
+            if frame_size and frame_size[0]
+            else 1.0
+        )
+
+        camera = PerspectiveCamera(
+            position=camera_position.tolist(),
+            look_at=center.tolist(),
+            up="Y",
+            aspect=aspect,
+        )
+
+        scene = Scene(camera=camera, lights=[])
+        for mesh_obj in mesh_objects:
+            scene.add(mesh_obj)
+
+        scene.write(scene_path)
+
+        if update:
+            self.scene_path = scene_path
+
+        return scene_path
+
+
+class HumanPose2D(Label):
+    """2D human pose keypoints and detection for image samples.
+
+    This class wraps native FiftyOne label types (Keypoint and Detection) to
+    enable proper rendering in the FiftyOne App using the core overlay logic.
+    The detection's instance field can be used to link this 2D pose to the
+    corresponding 3D Person3D label in a grouped dataset.
+
+    Args:
+        pose (None): a :class:`Keypoint` instance containing the 2D keypoints
+        detection (None): a :class:`Detection` instance for the person bounding box.
+            The detection.instance field can be set to link to Person3D labels.
+    """
+
+    pose = fof.EmbeddedDocumentField(Keypoint)
+    detection = fof.EmbeddedDocumentField(Detection)
+
+    def __init__(
+        self,
+        pose=None,
+        detection=None,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.pose = pose
+        self.detection = detection
+
+
+class HumanPoses2D(_HasLabelList, Label):
+    """A list of 2D human poses in an image.
+
+    This wrapper class provides proper schema registration for lists of
```

Comment:
Woah does the App actually support visualizing boxes and keypoints in `HumanPoses2D` fields in this format today?

I didn't think it was smart enough to automatically traverse lists of embedded documents whose *attributes* are `Label` classes!